Draft-Accepted/Install-PowerShell.md

```diff
@@ -0,0 +1,217 @@
+- Provide advanced options with parameters;
+  - Specific version
+  - Stable, Preview or LTS
+  - Scope - location of installation based on permissions
+
+## Goals/Non-Goals
+
+Goals:
+
+- Be discoverable as other built-in cmdlets
+- By default, download and install the latest stable release to the users Scope
+  - AllUsers - if user has administrative privileges
+  - CurrentUser - if the user does not have administrative privileges
+- Allow users to use remoting for install at scale with Invoke-Command
+- Allow advanced users to specify the version of PowerShell
+- Allow advanced users to specify scope (location of installation)
+- Allow advanced users to specify channel Stable/Preview/LTS
+- Users should receive a warning to continue the installation (ConfirmImpact=High)
+
+Non-goals:
+
+- Removal of PowerShell (Remove-PowerShell7)
+- Updating of PowerShell (Update-PowerShell7)
+- Disconnect scenarios - Users that are air-gapped may download the appropriate release from GitHub.
+- Handling of additional MSI switch options - i.e. Enable PSRemoting
+
+## Specification
+
+Cmdlet name: Install-PowerShell7
+
+Alias: None
+
+The default behavior is to download and extract the .ZIP package, then install the latest stable
+release of PowerShell into the user scope allowed by permissions. If administrative privilege,
+install to the scope of AllUsers. If user does not have administrative privilege, install to the
+scope CurrentUser.
+
+- The package install location - c:\temp
+- The installer should update the Users PATH for pwsh
+- The installer should enable Microsoft Updates for scope AllUsers. Due to permission requirements,
+  the scope CurrentUser will require manual updating.
+- Confirm impact should be set to **High** to force a warning and the ability to abort
+
+The following parameters can be added by experienced PowerShell users to customize the installation.
+
+- Parameter: Version - overrides the default and allows the user to specify the version of
+  PowerShell that will be installed. PowerShell version 7.0 and 7.1 are out of support, so version
+  numbers should begin with 7.2
+- Parameter: Scope - overrides the default and allows the user to specify the installation location
+  (scope) AllUsers or CurrentUser.
+- Parameter: Channel - options Stable/Preview/LTS - allows the user to select if they want the
+  current preview or Long Term Service (LTS) release
+- Parameter: Passthru - to provide informative output post installation
+
+## Syntax and Parameter Sets
+
+### ParameterSet Version (Default)
+
+```
+Install-PowerShell7 [[-Version] <string>] [-Scope {AllUsers | CurrentUsers}] [-Passthru] [-WhatIf] [-Confirm]  [<CommonParameters>]
+```
+
+### ParameterSet Channel
+
+```
+Install-PowerShell7 -Channel {Preview | LTS | Stable} [-Scope {AllUsers | CurrentUser}] [-Passthru] [-WhatIf] [-Confirm]
+[<CommonParameters>]
+```
+
+The **Version** parameter has the following validation attribute: [ValidatePattern("^7\.\d+")]
```

Comment:
This means the regex would need to change if/when we get Pwsh 8. Should this be ANY valid version number? And allow the script to fail gratefully if a specified version does not exist on GitHub?

Reply:
I would prefer to try any version number and then fail gracefully if it can't download. 
Also **only** use validate pattern on PowerShell 5 **if** you think this message is OK: 
```
The argument "7" does not match the "^7\.\d+" pattern. Supply an argument that matches "^7\.\d+" and try the command again. 
```

Reply:
This cmdlet is intended to install only Powershell 7.